### PR TITLE
Add `graphs.enable_dynamic_create_drop` option

### LINKED ISSUE #1808: [Feature] Add `graphs.enable_dynamic_create_drop` option
### Feature Description (功能描述)

Add `DYNAMIC_CREATE_GRAPH` option to make creating graph more flexible.

### PR DESCRIPTION
**Pull request by simon824**
closed #1808

### CODE REVIEW

**Comment by simon824** [comment]
I have read the CLA Document and I hereby sign the CLA

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 1]
```diff
@@ -250,4 +250,12 @@ public static synchronized ServerOptions instance() {
                     null,
                     "hugegraph"
             );
+
+    public static final ConfigOption<Boolean>  DYNAMIC_CREATE_GRAPH =
+            new ConfigOption<>(
+                    "dynamic_create_graph",
+                    "Whether to create graph dynamically",
+                    disallowEmpty(),
+                    true
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:259]: Thank you for your contribution.
Not sure in what scenario the option needs to be turned off?

Comment (simon824) [reply hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:259]: Hi @javeme , thanks for your reply, I am worried that my online stable graph instance will be deleted by misoperation.

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 2]
```diff
@@ -250,4 +250,12 @@ public static synchronized ServerOptions instance() {
                     null,
                     "hugegraph"
             );
+
+    public static final ConfigOption<Boolean>  DYNAMIC_LISTEN_GRAPH_CHANGES =
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:254]: there are 2 spaces in ">  DYNAMIC"

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 3]
```diff
@@ -250,4 +250,12 @@ public static synchronized ServerOptions instance() {
...
+            new ConfigOption<>(
+                    "dynamic_create_graph",
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:256]: can we rename to "graphs.disable_dynamic_create_drop" and update the description?

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java (hunk 1 above)
Comment (javeme) [reply hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:259]: Get it, sounds make sense

File: hugegraph-api/src/main/java/com/baidu/hugegraph/core/GraphManager.java [hunk 4]
```diff
@@ -157,6 +159,10 @@ public HugeGraph cloneGraph(String name, String newName,
     }
 
     public HugeGraph createGraph(String name, String configText) {
+        E.checkArgument(hugeConfig.get(ServerOptions.ENABLE_DYNAMIC_CREATE_DROP),
+                        "Not allowed to create graph dynamically, " +
+                                "please set `enable_dynamic_create_drop` to true.",
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/core/GraphManager.java:164]: align with ""Not allowed..."

File: hugegraph-api/src/main/java/com/baidu/hugegraph/core/GraphManager.java [hunk 5]
```diff
@@ -157,6 +159,10 @@ public HugeGraph cloneGraph(String name, String newName,
...
+                        name);
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/core/GraphManager.java:165]: need to fill graph name parameter: "Not allowed to create graph dynamically: '%s', "?

File: hugegraph-api/src/main/java/com/baidu/hugegraph/core/GraphManager.java [hunk 6]
```diff
@@ -171,6 +177,10 @@ public HugeGraph createGraph(String name, String configText) {
 
     public void dropGraph(String name) {
         HugeGraph graph = this.graph(name);
+        E.checkArgument(hugeConfig.get(ServerOptions.ENABLE_DYNAMIC_CREATE_DROP),
+                        "Not allowed to drop graph dynamically, " +
+                                "please set `enable_dynamic_create_drop` to true.",
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/core/GraphManager.java:182]: align with ""Not allowed..."

File: hugegraph-api/src/main/java/com/baidu/hugegraph/core/GraphManager.java [hunk 7]
```diff
@@ -171,6 +177,10 @@ public HugeGraph createGraph(String name, String configText) {
...
+                        name);
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/core/GraphManager.java:183]: ditto

File: hugegraph-api/src/main/java/com/baidu/hugegraph/core/GraphManager.java [hunk 8]
```diff
@@ -157,6 +159,10 @@ public HugeGraph cloneGraph(String name, String newName,
     }
 
     public HugeGraph createGraph(String name, String configText) {
+        E.checkArgument(hugeConfig.get(ServerOptions.ENABLE_DYNAMIC_CREATE_DROP),
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/core/GraphManager.java:162]: can rename hugeConfig to conf, and expect 'this.xx' prefix style for member access: "this.conf"

**Review by javeme — APPROVED** [review]

**Review by imbajin — APPROVED** [review]

**Pull request merged**